File: client/src/main/java/au/com/codeka/warworlds/client/net/ServerUrl.java
package au.com.codeka.warworlds.client.net;

import java.net.URI;
import java.net.URISyntaxException;

import au.com.codeka.warworlds.client.BuildConfig;
import au.com.codeka.warworlds.client.util.GameSettings;
import au.com.codeka.warworlds.client.util.Version;

/**
 * Helper class for getting the current server URL.
 */
public class ServerUrl {
  public static String getUrl() {
    return GameSettings.i.getString(GameSettings.Key.SERVER);
  }

  public static String getHost() {
    try {
      URI uri = new URI(getUrl());
      return uri.getHost();
    } catch (URISyntaxException e) {
      // Shouldn't happen.
      return null;
    }
  }

  public static String getUrl(String path) {
    if (path.startsWith("/")) {
      path = path.substring(1);
    }
    return getUrl() + path;
  }
}

File: client/src/main/kotlin/au/com/codeka/warworlds/client/net/ServerUrl.kt
package au.com.codeka.warworlds.client.net

import au.com.codeka.warworlds.client.util.GameSettings
import java.net.URI
import java.net.URISyntaxException

/** Helper class for getting the current server URL. */
object ServerUrl {
  val url: String
    get() = GameSettings.getString(GameSettings.Key.SERVER)

  val host: String?
    get() = try {
      val uri = URI(url)
      uri.host
    } catch (e: URISyntaxException) {
      // Shouldn't happen.
      null
    }

  fun getUrl(path: String): String {
    var p = path
    if (p.startsWith("/")) {
      p = p.substring(1)
    }
    return url + p
  }
}
